### PR TITLE
Report uncaught exceptions to DS

### PR DESCRIPTION
**Pull request by Twinters007**
So I think this will work, but I'm not completely sure about the inner workings of wpilib to know if this is the right place to put this. 

I also don't know how to test for this, since it involves reporting to the DS

### CODE REVIEW

File: wpilib/wpilib/_impl/main.py [hunk 1]
```diff
@@ -9,6 +9,12 @@
 from .logconfig import configure_logging
 import hal_impl
 
+from wpilib.driverstation import DriverStation
+
+def my_except_hook(exctype, value, traceback):
+    DriverStation.reportError("ERROR Unhandled {}".format(exctype), True)
+    sys.__excepthook__(exctype, value, traceback)
+sys.excepthook = my_except_hook
```
Comment (virtuald) [inline wpilib/wpilib/_impl/main.py]: Hm, seems like you would want to do this only when the code is actually launched, instead of always.

Comment (Twinters007) [reply wpilib/wpilib/_impl/main.py]: you mean if 'run' is the entry point or whatever?

Comment (virtuald) [reply wpilib/wpilib/_impl/main.py]: Well. Maybe only when the user's code is about to be launched?

**Comment by Twinters007** [comment]
How about now @virtuald

**Comment by virtuald** [comment]
Turns out, this isn't actually useful:

* Our main thread always catches exceptions in RobotBase
* `sys.excepthook` doesn't catch exceptions that arise in other threads (and this is a particularly annoying problem to solve -- see https://bugs.python.org/issue1230540)

**Pull request merged**